src/mpi/comm/comm_impl.c

```diff
@@ -678,6 +678,72 @@ int MPIR_Comm_dup_with_info_impl(MPIR_Comm * comm_ptr, MPIR_Info * info, MPIR_Co
     goto fn_exit;
 }
 
+int MPIR_Comm_create_from_group_impl(MPIR_Group * group_ptr, const char *stringtag,
+                                     MPIR_Info * info_ptr, MPIR_Errhandler * errhan_ptr,
+                                     MPIR_Comm ** p_newcom_ptr)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    /* This implementation assumes an internal MPI_COMM_WORLD already exist.
+     *
+     * Refer to TODOs below inside the branches:
+     * The next implementation will relax but assume the first call to this function will
+     * either from a "world" pset or "self" pset, during former a comm world will be created.
+     *
+     * The final implementation will further relax and allow partial initializations.
+     */
+
+    /* Need convert stringtag to tag, use macro from uthash.h to calculate hash value */
+    unsigned hash;
+    int n = strlen(stringtag);
+    HASH_VALUE(stringtag, n, hash);
+
+    /* NOTE: tag will be used with MPIR_TAG_COLL_BIT on, ref. MPIR_Get_contextid_sparse_group */
+    int tag = hash % (MPIR_Process.attrs.tag_ub);
+
+    if (MPIR_Process.comm_world) {
+        if (!MPIR_Process.comm_world->local_group) {
+            /* need create local_group for MPIR_Comm_create_group_impl to work */
+            MPIR_Group *world_group_ptr;
+            mpi_errno = MPIR_Comm_group_impl(MPIR_Process.comm_world, &world_group_ptr);
+            MPIR_ERR_CHECK(mpi_errno);
+            MPIR_Group_release(world_group_ptr);
+        }
```

Comment:
The comment is OK. But `MPIR_Comm_group_impl ` is not really for this purpose. So I feel weird to use it here. Might be better if we define a specific routine to explicitly initialize `local_group`. We can use the routine here and  in `MPIR_Comm_group_impl`